ray-operator/apis/ray/v1alpha1/raycluster_types.go

```diff
@@ -110,7 +118,16 @@ type RayClusterStatus struct {
 	MaxWorkerReplicas int32 `json:"maxWorkerReplicas,omitempty"`
 	// LastUpdateTime indicates last update timestamp for this cluster status.
 	// +nullable
-	LastUpdateTime *metav1.Time `json:"lastUpdateTime,omitempty"`
+	LastUpdateTime      *metav1.Time  `json:"lastUpdateTime,omitempty"`
+	HeadStatus          GroupStatus   `json:"headStatus,omitempty"`
+	WorkerGroupStatuses []GroupStatus `json:"workerGroupStatuses,omitempty"`
+}
+
+type GroupStatus struct {
+	// Name of the group. Can be empty in the case of the head "group".
+	GroupName string `json:"groupName,omitempty"`
+	// DetectedRayResources is the map of resource name to Ray resource capacity for the members of the head or worker group.
+	DetectedRayResources RayResources `json:"detectedRayResources,omitempty"`
```

Comment:
@sriram-anyscale I wonder if you have an idea how to do this better. My understanding is this: We have this field, so that we can put the full ray resources in here (computed from the rayResources annotations as well as the pod limits by the KubeRay operator), which are then read by the python autoscaler process -- this is to avoid re-computing the same computation in the python code. It seems odd to me that we need to introduce a (user visible) status to make this happen, maybe there is a better way :)

Reply:
Thinking about it a second time, I think it would be better to just have the autoscaler redo the computation in Python. The logic is not that complex and it would be validated by unit tests on both the Ray Python side and the KubeRay Golang side.

Just the price paid for using different programming languages.

For the Ray dashboard's logic for detecting resources inside a container, I literally ported the Docker Go code into Python...

Reply:
Hmm, actually @akanso suggestion of something less mysterious `autoscalingHints` sounds good to me as well.

As a bonus, the groupStatus field could be extended to carry more information in the future, see https://github.com/ray-project/kuberay/issues/223

Reply:
Or `autoscalerResourceHints` for something more specific.

Reply:
Adding user-visible status actually has the advantage that it is _user-visible_. (The user can see what the system is doing.)

Reply:
I keep vacillating on this (maybe because it's pretty immaterial)

I think it'd be better to just have the autoscaler re-do the computation and keep the outward interface simple.

Reply:
Thankfully, the K8s Python client provides a utility for this: https://github.com/kubernetes-client/python/blob/master/kubernetes/utils/quantity.py

Reply:
I think we should look at the general flow of information/control:
- So far it was the user or autoscaler that updates the spec and the Kuberay operator acts on these updates in its reconciler loop; the status field is updated for informational purposes only
- It seems like now the Kuberay operator is reading the spec and updating the status (or adding a hints file in spec?) for consumption by the autoscaler. (but let me read the code in more detail)

What Kuberay is doing is just a simple computation on primary info on the spec and updating the spec or status. Until this happens, the autoscaler can't use this information - which seems wrong and circular. I just feel it is best for the autoscaler to also do this simple computation (yes two languages is the main issue here).

We can chat more as necessary.